spec/client/config_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "client/config"
+
+RSpec.describe Client::Config do # rubocop:disable Metrics/BlockLength
+  let(:config) { Client::Config.new }
+
+  describe "#initialize" do
+    it "initializes with default values" do
+      expect(config.app_id).to be_nil
+      expect(config.private_key).to be_nil
+      expect(config.root_path_to_save).to be_nil
+      expect(config.site).to eq("https://localize.cirro.io/api")
+      expect(config.api_version).to eq("v2")
+    end
+  end
+
+  describe "#validate!" do # rubocop:disable Metrics/BlockLength
+    context "when all required attributes are set" do
+      before do
+        config.app_id = "test_app_id"
+        config.private_key = "test_private_key"
+        config.root_path_to_save = "/path/to/save"
+      end
+
+      it "does not raise any error" do
+        expect { config.validate! }.not_to raise_error
+      end
+    end
+
+    context "when required attributes are missing" do
+      it "raises a NameError with appropriate message" do
+        expect do
+          config.validate!
+        end.to raise_error(NameError,
+                           "Error: uninitialized constant APP_ID in .env, Error: uninitialized constant PRIVATE_KEY in"\
+                           " .env, Error: uninitialized constant ROOT_PATH_TO_SAVE in .env")
```

Comment:
outdated errors about .env 🫠
will fix the wording to an undefined config variable.